src/main/webapp/app/pages/variantRecommendation/MutationDetailedFrequencyTab.tsx

```diff
@@ -0,0 +1,187 @@
+import TwoRangeSlider from 'app/oncokb-commons/components/twoRangeSlider/TwoRangeSlider';
+
+function MutationDetailedFrequencyTab(props) {
+  const [cancerType, SetCancerType] = useState('');
+
+  const detailedFre = useMemo(() => {
+    if (props.data) {
+      const detailedFrequencyDataList: DetailedFrequencyData[] = [];
+      props.data.map(item => {
+        const detailedFrequencyData: DetailedFrequencyData = {
+          CANCER_TYPE: item.CANCER_TYPE,
+          CANCER_TYPE_DETAILED: item.CANCER_TYPE_DETAILED,
+          Hugo_Symbol: item.Hugo_Symbol,
+          Entrez_Gene_Id: item.Entrez_Gene_Id,
+          HGVSp_Short: item.HGVSp_Short,
+          Number: Number(item.Number),
+          Percentage: Number(item.Percentage),
+        };
+        detailedFrequencyDataList.push(detailedFrequencyData);
+      });
+      return detailedFrequencyDataList;
+    } else {
+      return [];
+    }
+  }, [props.data]);
+
+  const cancerOptions = useMemo(() => {
+    const uniqueCancerTypes = [...new Set(detailedFre.map(item => item.CANCER_TYPE))];
+    return [
+      <option key="" value="">
+        All
+      </option>,
+      ...uniqueCancerTypes.map(item => (
+        <option key={item} value={item}>
+          {item}
+        </option>
+      )),
+    ];
+  }, [detailedFre]);
+
+  const detailedCancerOptions = useMemo(() => {
+    const uniqueDetailedCancerTypes = [
+      ...new Set(detailedFre.filter(item => item.CANCER_TYPE === cancerType).map(item => item.CANCER_TYPE_DETAILED)),
+    ];
+    return [
+      <option key="" value="">
+        All
+      </option>,
+      ...uniqueDetailedCancerTypes.map(item => (
+        <option key={item} value={item}>
+          {item}
+        </option>
+      )),
+    ];
+  }, [cancerType]);
+
+  const columns: SearchColumn<DetailedFrequencyData>[] = [
+    {
+      accessor: 'CANCER_TYPE',
+      Header: 'Cancer Type',
+      Filter({ filter, onChange }) {
+        return (
+          <Input
+            type="select"
+            value={filter ? filter.value : ''}
+            onChange={event => {
+              onChange(event.target.value);
+              SetCancerType(event.target.value);
+            }}
+          >
+            {cancerOptions}
+          </Input>
+        );
+      },
+      filterMethod(filter, row) {
+        if (!filter.value) return true;
+        return String(row[filter.id]).toLowerCase().includes(filter.value.toLowerCase());
+      },
+    },
+    {
+      accessor: 'CANCER_TYPE_DETAILED',
+      Header: 'Detailed Cancer Type',
+      Filter({ filter, onChange }) {
+        return (
+          <Input type="select" value={filter ? filter.value : ''} onChange={event => onChange(event.target.value)}>
+            {detailedCancerOptions}
+          </Input>
+        );
+      },
+      filterMethod(filter, row) {
+        if (!filter.value) return true;
+        return String(row[filter.id]).toLowerCase().includes(filter.value.toLowerCase());
```

Comment:
See if you can turn this into a function. I see the same repeated several times.